specify data type of websocket message event (#22695)
mdn/content: files/en-us/web/api/websocket/message_event/index.md

_In addition to the properties listed below, properties from the parent interface, {{domxref("Event")}}, are available._

- {{domxref("MessageEvent.data", "data")}} {{ReadOnlyInline}}

>   - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}.

Reviewer: ```suggestion
  - : The data sent by the message emitter. The type of this property depends on the type of the WebSocket message and the value of {{domxref("WebSocket.binaryType")}}.
```
...but how do we know the type of the WebSocket message? It seems to be visible as the `opcode` value in https://datatracker.ietf.org/doc/html/rfc6455#section-5.2, but I couldn't see how this is visible in the JS API.

Reviewer 2: I'm not sure who is "we" in the above; as a developer, you would know by e.g. testing `typeof data === 'string'` (and if it isn't, then you would check `binaryType` to figure how the binary data is represented). The browser is responsible for mapping the protocol bits to the JS surface.

(in reality, in most cases there is a tight coupling between the JS consumer and the emitter, so the developer may simply know the type from out of band coordination)

Reviewer: Thank you @dontcallmedom , of course, that makes sense.

>   - : The data sent by the message emitter. The type of this property depends on the type of the WebSocket message and the value of {{domxref("WebSocket.binaryType")}}.

Reviewer 3: Could you move this information to the `data` page? This description here should only be a one-line overview.

Reviewer: @jingkaimori , could you also make this change please? Then I think we should be able to merge this PR.

>     - If the received socket message is a string, then this field is a string.
>     - If the received socket message is binary type, then the type of this field will be inferred from `binaryType` of this socket:

Reviewer: ```suggestion
    - If the message type is "text", then this field is a string.
    - If the message type is "binary" type, then the type of this property can be inferred from the `binaryType` of this socket:
```

_In addition to the properties listed below, properties from the parent interface, {{domxref("Event")}}, are available._

- {{domxref("MessageEvent.data", "data")}} {{ReadOnlyInline}}

>   - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}
>     - if received socket message is string type, then this field is a {{jsxref("String")}}.
>     - if received socket message is binary type, then type of this field will be inferred from `binaryType` of this socket.
>       - {{jsxref("ArrayBuffer")}} if `binaryType` is `"arraybuffer"`,
>       - {{domxref("Blob")}} if `binaryType` is `"blob"`.

Reviewer 3: ```suggestion
  - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}.
    - If the received socket message is a string, then this field is a string.
    - If the received socket message is binary type, then the type of this field will be inferred from `binaryType` of this socket:
      - {{jsxref("ArrayBuffer")}} if `binaryType` is `"arraybuffer"`,
      - {{domxref("Blob")}} if `binaryType` is `"blob"`.
```
